docs/source/design/multiprocessing.md

```diff
@@ -0,0 +1,185 @@
+# Python Multiprocessing
+
+*Note that source code references are to the state of the code at the time of writing in December, 2024.*
```

Comment:
I think we should add a `TLDR` at the top which says if you encounter the following error:

```bash
RuntimeError:
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.
        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:
            if __name__ == '__main__':
                freeze_support()
                ...
        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.
        To fix this issue, refer to the "Safe importing of main module"
        section in https://docs.python.org/3/library/multiprocessing.html
```

Then guard your import with `if __name__ == "__main__"`

Reply:
I updated the user-oriented "Debugging Tips" page with a new section on Python multiprocessing. It lists the errors / warnings someone might see and gives instructions on how to use the main guard to solve them.

I updated the dev-oriented design doc with a link at the top over to the debugging page, as well.